lumibot/strategies/strategy_executor.py

```diff
@@ -161,8 +161,23 @@ def sync_broker(self):
                                 f"to be {obroker} because what we have in memory does not match the broker."
                             )
                 else:
-                    # Add to order in lumibot.
-                    self.broker._process_new_order(order)
+                    # If it is the brokers first iteration then fully process the order because it is likely
+                    # that the order was filled/canceled/etc before the strategy started.
+                    if self.broker._first_iteration:
+                        if order.status == Order.OrderStatus.FILLED:
```

Comment:
the order object already has an "is_filled()" and "is_cancelled()" function available that you can use.  It is cleaner and easier to read.

Reply:
honestly i dont think this would be cleaner. it would then be a mix of is_filled(), is_cancelled(), Order.OrderStatus.PARTIALLY_FILLED and Order.OrderStatus.NEW